Jenkinsfile

```diff
@@ -92,56 +92,136 @@ pipeline {
+        * It then stashes the tokens, which are unstashed in the stage that runs the
+        * GCP Authenticator tests using the tokens.
+        * This way we can have a light-weight GCE instance that has no need for conjurops
+        * or git identities and is not open for SSH
+        */
+        stage('GCP - Stage A - Compute Engine') {
           steps {
+            echo '-- Allocating Google Compute Engine'
             script {
+              dir('ci/authn-gcp') {
+                stash name: 'get_gce_tokens_script', includes: 'get_gce_tokens_to_files.sh'
+              }
               node('executor-v2-gcp-small') {
+                echo '-- Google Compute Engine allocated'
+                echo '-- Get Google Cloud project name from Google metadata server'
+                env.GCP_PROJECT = sh (
+                  script: 'curl -s -H "Metadata-Flavor: Google" \
+                  "http://metadata.google.internal/computeMetadata/v1/project/project-id"',
+                  returnStdout: true
+                ).trim()
+                echo "Google Cloud project name: $GCP_PROJECT"
+                unstash 'get_gce_tokens_script'
                 sh '''
-                  retrieve_token() {
-                    local token_format="$1"
-                    local audience="$2"
-
-                    curl \
-                      -s \
-                      -H 'Metadata-Flavor: Google' \
-                      "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity?format=${token_format}&audience=${audience}"
-                  }
-
-                  echo "$(retrieve_token "full" "conjur/cucumber/host/test-app")" > "gcp_token_valid"
-                  echo "$(retrieve_token "full" "conjur/cucumber/host/non-existing")" > "gcp_token_non_existing_host"
-                  echo "$(retrieve_token "full" "conjur/cucumber/host/non-rooted/test-app")" > "gcp_token_non_rooted_host"
-                  echo "$(retrieve_token "full" "conjur/cucumber/test-app")" > "gcp_token_user"
-                  echo "$(retrieve_token "full" "conjur/non-existing/host/test-app")" > "gcp_token_non_existing_account"
-                  echo "$(retrieve_token "full" "invalid_audience")" > "gcp_token_invalid_audience"
-                  echo "$(retrieve_token "standard" "conjur/cucumber/host/test-app")" > "gcp_token_standard_format"
+                ./get_gce_tokens_to_files.sh || exit 1
                 '''
+                stash name: 'authnGceTokens', includes: 'gcp_token_*', allowEmpty:false
+              }
+            }
+          }
+          post {
+           failure {
+            script {
+              env.GCP_ENV_ERROR = "true"
+            }
+           }
+           success {
+            script {
+              env.GCE_TOKENS_FETCHED = "true"
+            }
+            echo '-- Finished fetching GCE tokens.'
+           }
+          }
+        }
+        stage('GCP - Stage B - Google Function') {
```

Comment:
Is there a reason you chose letters instead of numbers? Ex: Stage B instead of Stage 2

Reply:
Just wanted it to keep the order in the pipeline ui in the browser

Reply:
i wouldn't use the term `Stage <letter>` as it runs in parallel. I think that `GCP Authenticator preparation - Allocate GCE Instance` and `GCP Authenticator preparation - Allocate Google Function` is better

Reply:
I'll take it